homeassistant/components/modbus/validators.py

```diff
@@ -136,9 +135,8 @@ def struct_validator(config: dict[str, Any]) -> dict[str, Any]:
             )
             raise vol.Invalid(error)
 
-    if swap_type != CONF_SWAP_NONE:
+    if swap_type is not None:
```

Comment:
```suggestion
    if swap_type:
```

Should be fine as an empty string is not allowed

Reply:
ups !! corrected.